task/06-conditions-n-loops-tasks.js

```diff
@@ -296,7 +394,38 @@ function getDigitalRoot(num) {
  *   '{[(<{[]}>)]}' = true 
  */
 function isBracketsBalanced(str) {
-    throw new Error('Not implemented');
+    let arrTemp = [],
+        openBrackets = ['[', '(', '<', '{'],
+        closeBrackets = [']', ')', '>', '}'],
+        closeIndex,
+        openIndex,
+        len = str.length;
+
+    for (let i = 0;  i < len; i++) 
+    {
+        openIndex = openBrackets.indexOf(str[i]);
+        if (openIndex !== -1) 
+        {
+            arrTemp.push(openIndex);
+        }
+
+        closeIndex = closeBrackets.indexOf(str[i]);
+        if (closeIndex !== -1) 
+        {
+            openIndex = arrTemp.pop();
+            if (closeIndex !== openIndex) 
+            {
+                return false;
+            }
+        }
+    }
+
+    if (arrTemp.length != 0) 
```

Comment:
Here you can return the result of a boolean expression.

Reply:
Fixed.